### PR TITLE
t10 api art queries

### PR DESCRIPTION
**Pull request by lanc2112b**
Add sort, order query options.
Add topic filter.  
Refactor selectUsernameByName to correct model.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
...
+          expect(articles).toBeInstanceOf(Array);
+          expect(articles).toHaveLength(1); // 1 cat article
+          expect(articles).toBeSorted({ key: "created_at", descending: true }); //default
+          articles.forEach((article) => {
+            expect(article.author).toBe('rogersop');
+            expect(article.topic).toBe("cats");
+            expect(article).toMatchObject({
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: expect.any(String),
```
Comment (emilyb93) [inline __tests__/app.test.js:147]: I would say we know that its going to be a topic of 'cats'

File: __tests__/app.test.js [hunk 2]
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
...
+            });
+          });
+        });
+    });
+  });
+
+  describe("GET /api/articles (10) filtered by queries", () => {
+    it("200: Returns an array of objects containing articles filtered by topic > mitch", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch") // has one cat article
```
Comment (emilyb93) [inline __tests__/app.test.js]: I think someones been copying their code hmmmm

File: __tests__/app.test.js [hunk 3]
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
...
+        .expect(200)
+        .then((response) => {
+          const { articles } = response.body;
+          expect(articles).toBeInstanceOf(Array);
+          expect(articles).toHaveLength(11); // 11 mitch articles
+          expect(articles).toBeSorted({ key: "votes", ascending: true });
+          articles.forEach((article) => {
+            expect(article).toMatchObject({
+              author: expect.any(String),
+              topic: expect.any(String),
```
Comment (emilyb93) [inline __tests__/app.test.js]: We've requested specifically the topic of 'mitch' so i should expect that they all should have that topic

File: __tests__/app.test.js (hunk 1 above)
Comment (lanc2112b) [reply __tests__/app.test.js:147]:
```expect(article.author).toBe('rogersop');
            expect(article.topic).toBe("cats");
```

File: __tests__/app.test.js [hunk 4]
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
...
+              article_img_url: expect.any(String),
+              comment_count: expect.any(String),
+            });
+          });
+        });
+    });
+  });
+
+
+
```
Comment (emilyb93) [inline __tests__/app.test.js:289]: Great testing so far, can i just see a test to prove that if i pass order=desc it still works.
Also do you think your code can handle both desc and DESC, will this matter to your api, will your client  have to make this distinction?

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Just add a test to prove that order=desc is okay to do and have a look at my other comments and then its fine to merge

**Pull request merged**